core/src/blocktree.rs

```diff
@@ -759,21 +755,20 @@ impl Blocktree {
 
     pub fn set_root(&self, slot: u64) -> Result<()> {
         let mut root_slot = self.root_slot.write().unwrap();
-        let mut db = self.db.write().unwrap();
+        let _lock = self.batch_processor.write().unwrap();
```

Comment:
Think we can move the root flag into a separate column family so that set_root and insert_data_blob don't have to compete for locks either?

Reply:
yep, isn't really meta-specific, just need it persisted